pubsub/google/cloud/pubsub_v1/subscriber/message.py

```diff
@@ -151,7 +153,11 @@ def publish_time(self):
         Returns:
             datetime: The date and time that the message was published.
         """
-        return self._message.publish_time
+        timestamp = self._message.publish_time
+        delta = datetime.timedelta(
+            seconds=timestamp.seconds,
+            microseconds=timestamp.nanos // 1000)
```

Comment:
Instead of removing precision, should this be a DatetimeWithNanoseconds as defined in https://github.com/GoogleCloudPlatform/google-cloud-python/pull/4979?

Reply:
@tswast I considered it.  The issue is that there is no clean / efficient way to create an instance of `DateTimeWithNanoseconds` given only a delta:  there is no corresponding `TimeDeltaWIthNanoseconds`, and it doesn't override `__add__` etc. anyway. That class is defined in `api_core`, which we have been somewhat loath to bump.

To work around that deficit here, we would need to create a "normal" `datetime.datetime` first, then use its attributes to generate the constructor arguments for the `DateTimeWithNanoseconds` instance.  Does that seem worthwhile here?  I'm unclear how the field will be used.

Reply:
SGTM to use normal datetime. If it becomes an issue in the future we can always use DateTimeWithNanoseconds then.

Publish time could be used to order events, but it's not as important here as it would be for Spanner. In Spanner the nanosecond precision is important because it is used for consistency of writes.

Reply:
I don't think we use `DatetimeWithNanos` for the snapshot consistency bits in spanner:  it only gets used for marshalling parameters and column values.